sklearn/metrics/cluster/_unsupervised.py

```diff
@@ -175,19 +175,19 @@ def silhouette_samples(X, labels, *, metric='euclidean', **kwds):
 
     Parameters
     ----------
-    X : array [n_samples_a, n_samples_a] if metric == "precomputed", or, \
-             [n_samples_a, n_features] otherwise
-        Array of pairwise distances between samples, or a feature array.
+    X : array-like of shape (n_samples_a, n_samples_a) if metric == \
+            "precomputed", (n_samples_a, n_features) otherwise
```

Comment:
Same here:

```suggestion
    X : array-like of shape (n_samples_a, n_samples_a) if metric == \
            "precomputed" or (n_samples_a, n_features) otherwise
```